src/pipert2/core/base/pipe.py

```diff
@@ -52,8 +58,12 @@ def create_flow(self, flow_name: str, auto_wire: bool, *routines: Routine):
         self.flows[flow_name] = flow
 
         if auto_wire:
+            transmit = data_transmitter.transmit() if data_transmitter else self.default_data_transmitter.transmit()
+            receive = data_transmitter.receive() if data_transmitter else self.default_data_transmitter.receive()
```

Comment:
Could be modified to a single if block but its fine for now.
Or have a single variable with the desired data transmitter and getting its attributes from him.